lib/components/form/connected-links.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { LinkContainer } from 'react-router-bootstrap'
+import { Redirect } from 'react-router'
+import { Link } from 'react-router-dom'
+
+/**
+ * This function enhances the routing components imported above
+ * by preserving the itinerary search query from the redux state
+ * when redirecting the user between the main map and account-related pages,
+ * so that when the user returns to the map, the itinerary that was previously
+ * displayed is shown again.
+ * Implementers only need to specify the 'to' route and
+ * do not need to hook to redux store to retrieve the itinerary search query.
+ * @param RoutingComponent The routing component to enhance.
+ * @returns A new component that passes the redux search params to
+ *          the RoutingComponent's 'to' prop.
+ */
+const connectLink = RoutingComponent =>
```

Comment:
A better name for this connect method might be something like `connectToQueryParams`.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/280/commits/35eb848eeb28b878ba59d7fd07e01b18f92fa260.